rpc/client/client_test.go

```diff
@@ -242,6 +242,62 @@ func TestGetBlockByHash(t *testing.T) {
 	require.NoError(err)
 }
 
+func TestTx(t *testing.T) {
+	assert := assert.New(t)
+	require := require.New(t)
+
+	mockApp := &mocks.Application{}
+	mockApp.On("InitChain", mock.Anything).Return(abci.ResponseInitChain{})
+	key, _, _ := crypto.GenerateEd25519Key(crand.Reader)
+	node, err := node.NewNode(context.Background(), config.NodeConfig{
+		DALayer:    "mock",
+		Aggregator: true,
+		BlockManagerConfig: config.BlockManagerConfig{
+			BlockTime: 200 * time.Millisecond,
+		}},
+		key, proxy.NewLocalClientCreator(mockApp),
+		&tmtypes.GenesisDoc{ChainID: "test"},
+		log.TestingLogger())
+	require.NoError(err)
+	require.NotNil(node)
+
+	rpc := NewClient(node)
+	require.NotNil(rpc)
+	mockApp.On("BeginBlock", mock.Anything).Return(abci.ResponseBeginBlock{})
+	mockApp.On("EndBlock", mock.Anything).Return(abci.ResponseEndBlock{})
+	mockApp.On("Commit", mock.Anything).Return(abci.ResponseCommit{})
+	const evType = "ethereum_tx"
+	const evKey = "ethereumTxHash"
+	const evValue = "RLP encoded Keccak hash"
+	mockApp.On("DeliverTx", mock.Anything).Return(abci.ResponseDeliverTx{
+		Events: []abci.Event{{
+			Type: evType,
+			Attributes: []abci.EventAttribute{{
+				Key:   []byte(evKey),
+				Value: []byte(evValue),
+				Index: true,
+			}},
+		}},
```

Comment:
In this case, events are not required. Empty `ResponseDeliverTx` would be OK.
Could you also deliver one more Tx, and add a case for querying for non-existing Tx?

Reply:
Thanks for the feedback!

Will update this today and push up

Reply:
@tzdybal Do we not know how we want to handle this error here? https://github.com/celestiaorg/optimint/blob/main/state/txindex/kv/kv.go#L49

Could I update that method to return the err instead?

Reply:
Indexer code was borrowed from Tendermint, which is panicing (and recovering those panics) a lot. In Optimint, from start we mostly work with errors. I created issue for changing panics to erorrs (#279).

For now, I would just use `assert.Panics` to verify existing code.

Reply:
Ok perfect, will do as suggested

Reply:
@tzdybal  I did the following
```
		resTx, errTx := rpc.Tx(context.Background(), tx2.Hash(), true)
		assert.Nil(resTx)
		assert.Error(errTx)
```
to satisfy `golangci-lint` - I can't seem to see the same lint issue locally though